src/main/java/lotto/domain/LottoDescription.java

```diff
@@ -0,0 +1,60 @@
+package lotto.domain;
+
+import java.util.Arrays;
+import java.util.List;
+
+public enum LottoDescription {
+    NOMATCH(0, true, "NOMATCH", 0),
+    MATCH3(3, false, "MATCH3", 5000),
+    MATCH4(4, false, "MATCH4", 50000),
+    MATCH5(5, false, "MATCH5", 150000),
+    MATCHBONUS(5, true, "MATCHBONUS", 30000000),
+    MATCH6(6, false, "MATCH6", 2000000000);
+
+    private final int matchCount;
+    private final boolean isBonusNumber;
+    private final String hash;
+    private final int winning;
+
+    LottoDescription(int matchCount, boolean isBonusNumber, String hash, int winning) {
+        this.matchCount = matchCount;
+        this.isBonusNumber = isBonusNumber;
+        this.hash = hash;
+        this.winning = winning;
+    }
+
+    public static String findLottoHash(int matchCount, boolean isBonusNumber) {
+        return Arrays.stream(values())
+            .filter(lottoResult -> lottoResult.matchCount == matchCount)
+            .filter(lottoResult -> lottoResult.isBonusNumber == isBonusNumber)
+            .findAny()
+            .orElse(NOMATCH)
+            .hash;
+    }
+
+    public String getHash() {
+        return hash;
+    }
+
+    public int getWinning() {
+        return this.winning;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public boolean isBonusNumber() {
+        return isBonusNumber;
+    }
+
+    public static List<LottoDescription> toList() {
+        return Arrays.asList(
+            LottoDescription.MATCH3,
+            LottoDescription.MATCH4,
+            LottoDescription.MATCH5,
+            LottoDescription.MATCHBONUS,
+            LottoDescription.MATCH6
+        );
```

Comment:
만약 `LottoDescription`에 새로운 상수 값이 추가된다면 어떻게 될까요? 
아마도 지금의 상황에서는 코드가 한 줄 추가될거에요. 👀 

아래와 같이 변경해보면 어떨까요? 😃 
```suggestion
        return Arrays.stream(values())
                .collect(Collectors.toList());
```

Reply:
제안해주신 방법대로 변경하였습니다:)